<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>ADMIXTURE.jl · ADMIXTURE.jl</title><link href="https://fonts.googleapis.com/css?family=Lato|Roboto+Mono" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.11.1/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-dark.css" data-theme-name="documenter-dark"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit">ADMIXTURE.jl</span></div><form class="docs-search" action="search/"><input class="docs-search-query" id="documenter-search-query" name="q" type="text" placeholder="Search docs"/></form><ul class="docs-menu"><li class="is-active"><a class="tocitem" href>ADMIXTURE.jl</a><ul class="internal"><li><a class="tocitem" href="#Installation"><span>Installation</span></a></li><li><a class="tocitem" href="#Quick-example"><span>Quick example</span></a></li><li><a class="tocitem" href="#Detailed-options"><span>Detailed options</span></a></li><li><a class="tocitem" href="#Run-ADMIXTURE-directly"><span>Run ADMIXTURE directly</span></a></li><li><a class="tocitem" href="#References"><span>References</span></a></li></ul></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>ADMIXTURE.jl</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>ADMIXTURE.jl</a></li></ul></nav><div class="docs-right"><a class="docs-edit-link" href="https://github.com/OpenMendel/ADMIXTURE.jl/blob/master/docs/src/index.md" title="Edit on GitHub"><span class="docs-icon fab"></span><span class="docs-label is-hidden-touch">Edit on GitHub</span></a><a class="docs-settings-button fas fa-cog" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-sidebar-button fa fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a></div></header><article class="content" id="documenter-page"><h1 id="ADMIXTURE.jl"><a class="docs-heading-anchor" href="#ADMIXTURE.jl">ADMIXTURE.jl</a><a id="ADMIXTURE.jl-1"></a><a class="docs-heading-anchor-permalink" href="#ADMIXTURE.jl" title="Permalink"></a></h1><p>ADMIXTURE.jl is a Julia wrapper of the popular <a href="http://dalexander.github.io/admixture/">ADMIXTURE program</a> for estimating ancestry in a model-based manner from large autosomal SNP genotype data sets.</p><h2 id="Installation"><a class="docs-heading-anchor" href="#Installation">Installation</a><a id="Installation-1"></a><a class="docs-heading-anchor-permalink" href="#Installation" title="Permalink"></a></h2><p>ADMIXTURE.jl requires Julia v1.3 or later. See documentation for usage. It is not yet registered and can be installed, in the Julia Pkg mode, by</p><pre><code class="language-">(@v1.5) Pkg&gt; add https://github.com/OpenMendel/ADMIXTURE.jl</code></pre><p>The original ADMIXTURE software thus ADMIXTURE.jl only support Linux and MacOS.</p><pre><code class="language-julia">versioninfo()</code></pre><pre><code class="language-none">Julia Version 1.5.0
Commit 96786e22cc (2020-08-01 23:44 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.7.0)
  CPU: Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = code
  JULIA_NUM_THREADS = 4</code></pre><pre><code class="language-julia"># for use in this tutorial
using ADMIXTURE, DelimitedFiles, Glob, Tar</code></pre><h2 id="Quick-example"><a class="docs-heading-anchor" href="#Quick-example">Quick example</a><a id="Quick-example-1"></a><a class="docs-heading-anchor-permalink" href="#Quick-example" title="Permalink"></a></h2><p>ADMIXTURE.jl only exports one function <code>admixture</code>. The usage resembles the original ADMIXTURE software. </p><p>Let&#39;s first download a sample data set (in PLINK binary format) to the <code>/hapmap3</code> folder.</p><pre><code class="language-julia">isfile(&quot;hapmap3.tar&quot;) || 
download(&quot;http://dalexander.github.io/admixture/hapmap3-files.tar.gz&quot;, 
         joinpath(pwd(), &quot;hapmap3.tar&quot;))
isdir(&quot;hapmap3&quot;) ||
Tar.extract(&quot;hapmap3.tar&quot;, joinpath(pwd(), &quot;hapmap3&quot;))
readdir(glob&quot;hapmap3/hapmap3.*&quot;)</code></pre><pre><code class="language-none">4-element Array{String,1}:
 &quot;hapmap3/hapmap3.bed&quot;
 &quot;hapmap3/hapmap3.bim&quot;
 &quot;hapmap3/hapmap3.fam&quot;
 &quot;hapmap3/hapmap3.map&quot;</code></pre><p>To estimate ancestry from <span>$K=3$</span> populations, simply run</p><pre><code class="language-julia">P, Q = admixture(&quot;hapmap3/hapmap3.bed&quot;, 3);</code></pre><pre><code class="language-none">****                   ADMIXTURE Version 1.3.0                  ****
****                    Copyright 2008-2015                     ****
****           David Alexander, Suyash Shringarpure,            ****
****                John  Novembre, Ken Lange                   ****
****                                                            ****
****                 Please cite our paper!                     ****
****   Information at www.genetics.ucla.edu/software/admixture  ****

Random seed: 43
Point estimation method: Block relaxation algorithm
Convergence acceleration algorithm: QuasiNewton, 3 secant conditions
Point estimation will terminate when objective function delta &lt; 0.0001
Estimation of standard errors disabled; will compute point estimates only.


┌ Info: ADMIXTURE command:
│ `/Users/huazhou/.julia/artifacts/7d0732a6f20d0cf8c106525a4602fc824e4cff31/dist/admixture_macosx-1.3.0/admixture /Users/huazhou/.julia/dev/ADMIXTURE/docs/hapmap3/hapmap3.bed 3`
└ @ ADMIXTURE /Users/huazhou/.julia/dev/ADMIXTURE/src/ADMIXTURE.jl:57
┌ Info: Output directory: /Users/huazhou/.julia/dev/ADMIXTURE/docs
└ @ ADMIXTURE /Users/huazhou/.julia/dev/ADMIXTURE/src/ADMIXTURE.jl:58


Size of G: 324x13928
Performing five EM steps to prime main algorithm
1 (EM) 	Elapsed: 0.239	Loglikelihood: -4.38757e+06	(delta): 2.87325e+06
2 (EM) 	Elapsed: 0.239	Loglikelihood: -4.25681e+06	(delta): 130762
3 (EM) 	Elapsed: 0.239	Loglikelihood: -4.21622e+06	(delta): 40582.9
4 (EM) 	Elapsed: 0.238	Loglikelihood: -4.19347e+06	(delta): 22748.2
5 (EM) 	Elapsed: 0.238	Loglikelihood: -4.17881e+06	(delta): 14663.1
Initial loglikelihood: -4.17881e+06
Starting main algorithm
1 (QN/Block) 	Elapsed: 0.493	Loglikelihood: -3.94775e+06	(delta): 231058
2 (QN/Block) 	Elapsed: 0.5	Loglikelihood: -3.8802e+06	(delta): 67554.6
3 (QN/Block) 	Elapsed: 0.574	Loglikelihood: -3.83232e+06	(delta): 47883.8
4 (QN/Block) 	Elapsed: 0.672	Loglikelihood: -3.81118e+06	(delta): 21138.2
5 (QN/Block) 	Elapsed: 0.611	Loglikelihood: -3.80682e+06	(delta): 4354.36
6 (QN/Block) 	Elapsed: 0.568	Loglikelihood: -3.80474e+06	(delta): 2085.65
7 (QN/Block) 	Elapsed: 0.582	Loglikelihood: -3.80362e+06	(delta): 1112.58
8 (QN/Block) 	Elapsed: 0.61	Loglikelihood: -3.80276e+06	(delta): 865.01
9 (QN/Block) 	Elapsed: 0.574	Loglikelihood: -3.80209e+06	(delta): 666.662
10 (QN/Block) 	Elapsed: 0.679	Loglikelihood: -3.80151e+06	(delta): 579.49
11 (QN/Block) 	Elapsed: 0.613	Loglikelihood: -3.80097e+06	(delta): 548.156
12 (QN/Block) 	Elapsed: 0.648	Loglikelihood: -3.80049e+06	(delta): 473.565
13 (QN/Block) 	Elapsed: 0.571	Loglikelihood: -3.80023e+06	(delta): 258.61
14 (QN/Block) 	Elapsed: 0.643	Loglikelihood: -3.80005e+06	(delta): 179.949
15 (QN/Block) 	Elapsed: 0.683	Loglikelihood: -3.79991e+06	(delta): 146.707
16 (QN/Block) 	Elapsed: 0.625	Loglikelihood: -3.79989e+06	(delta): 13.1942
17 (QN/Block) 	Elapsed: 0.681	Loglikelihood: -3.79989e+06	(delta): 4.60747
18 (QN/Block) 	Elapsed: 0.585	Loglikelihood: -3.79989e+06	(delta): 1.50012
19 (QN/Block) 	Elapsed: 0.583	Loglikelihood: -3.79989e+06	(delta): 0.128916
20 (QN/Block) 	Elapsed: 0.578	Loglikelihood: -3.79989e+06	(delta): 0.00182983
21 (QN/Block) 	Elapsed: 0.567	Loglikelihood: -3.79989e+06	(delta): 4.33787e-05
Summary: 
Converged in 21 iterations (14.858 sec)
Loglikelihood: -3799887.171935
Fst divergences between estimated populations: 
	Pop0	Pop1	
Pop0	
Pop1	0.163	
Pop2	0.073	0.156	
Writing output files.</code></pre><p>The <code>P</code> matrix contains estimated allele frequencies of the inferred ancestral populations.</p><pre><code class="language-julia">P</code></pre><pre><code class="language-none">13928×3 Array{Float64,2}:
 0.99999   0.99999   0.99999
 0.946581  0.934992  0.901852
 0.989626  0.382598  0.918612
 0.973109  0.682057  0.907595
 0.678695  0.918927  0.129153
 0.99999   0.99999   0.99999
 0.99999   0.990119  0.99999
 0.841989  0.203466  0.851233
 0.967501  0.86069   0.622157
 0.870693  0.862778  0.842376
 0.99999   0.99999   0.99999
 0.279564  0.830299  0.224164
 0.777817  0.555361  0.950294
 ⋮                   
 0.746245  0.769682  0.753071
 0.7462    0.769957  0.752793
 0.752064  0.734527  0.753725
 0.745593  0.769889  0.754917
 0.749709  0.769826  0.755746
 0.717428  0.67002   0.702682
 0.745593  0.769889  0.754917
 0.745593  0.769889  0.754917
 0.749937  0.767978  0.755766
 0.99999   0.996706  0.99999
 0.978953  0.982035  0.926256
 0.99999   0.993413  0.992614</code></pre><p>The <code>Q</code> matrix contains the estimated ancestry fractions.</p><pre><code class="language-julia">Q</code></pre><pre><code class="language-none">324×3 Array{Float64,2}:
 1.0e-5    0.896321  0.103669
 0.009659  0.830876  0.159465
 0.05577   0.725441  0.21879
 1.0e-5    0.866447  0.133543
 0.029255  0.88897   0.081775
 0.009302  0.859576  0.131122
 1.0e-5    0.715624  0.284366
 0.013736  0.810352  0.175913
 1.0e-5    0.727122  0.272868
 0.03487   0.821125  0.144004
 1.0e-5    0.886316  0.113674
 0.011359  0.863532  0.125109
 1.0e-5    0.856018  0.143972
 ⋮                   
 1.0e-5    0.99998   1.0e-5
 1.0e-5    0.99998   1.0e-5
 1.0e-5    0.99998   1.0e-5
 1.0e-5    0.99998   1.0e-5
 1.0e-5    0.99998   1.0e-5
 1.0e-5    0.99998   1.0e-5
 1.0e-5    0.99998   1.0e-5
 1.0e-5    0.99998   1.0e-5
 1.0e-5    0.99998   1.0e-5
 1.0e-5    0.99998   1.0e-5
 1.0e-5    0.99998   1.0e-5
 0.004387  0.995603  1.0e-5</code></pre><p>The output files of ADMIXTURE are available in the working directory.</p><pre><code class="language-julia">readdir(glob&quot;hapmap3.3.*&quot;)</code></pre><pre><code class="language-none">2-element Array{String,1}:
 &quot;hapmap3.3.P&quot;
 &quot;hapmap3.3.Q&quot;</code></pre><p>We can read the output files by</p><pre><code class="language-julia">P = readdlm(&quot;hapmap3.3.P&quot;, &#39; &#39;, Float64)
Q = readdlm(&quot;hapmap3.3.Q&quot;, &#39; &#39;, Float64);</code></pre><h2 id="Detailed-options"><a class="docs-heading-anchor" href="#Detailed-options">Detailed options</a><a id="Detailed-options-1"></a><a class="docs-heading-anchor-permalink" href="#Detailed-options" title="Permalink"></a></h2><p>The <code>admixture</code> function wraps most options for running ADMIXTURE program. The keywords closely follow those of ADMIXTURE.</p><article class="docstring"><header><a class="docstring-binding" id="ADMIXTURE.admixture" href="#ADMIXTURE.admixture"><code>ADMIXTURE.admixture</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia">admixture(inputFile, K; kwargs...)</code></pre><p>Calculate the admixture estimates assuming <code>K</code> populations. Input file can be  <code>.bed</code> (together with <code>.bim</code>, binary PLINK format), <code>geno</code> (STRUCTURE format),  or <code>ped</code> (12-coding PLINK format). </p><p><strong>Keyword arguments</strong></p><ul><li><code>B</code>: compute standard errors by bootstrap; default is <code>nothing</code>.  </li><li><code>j</code>: number of threads to be used; default is <code>nothing</code>.</li><li><code>s</code>: random seed; integer or <code>:time</code>; default is <code>nothing</code>. </li><li><code>C</code>: major convergence criterion; float or integer; default is <code>nothing</code>. </li><li><code>c</code>: minor convergence criterion; float or integer; default is <code>nothing</code>.</li><li><code>qn</code>: secant pairs to be used in quasi-Newton acceleration; default is 3.</li><li><code>l</code>: tuning parameter in penalized estimation.</li><li><code>e</code>: tuning parameter in penalized estimation.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/OpenMendel/ADMIXTURE.jl/blob/417043ef7e14ddc018798c25c5bf2f40b4146721/src/ADMIXTURE.jl#L16-L32">source</a></section></article><h2 id="Run-ADMIXTURE-directly"><a class="docs-heading-anchor" href="#Run-ADMIXTURE-directly">Run ADMIXTURE directly</a><a id="Run-ADMIXTURE-directly-1"></a><a class="docs-heading-anchor-permalink" href="#Run-ADMIXTURE-directly" title="Permalink"></a></h2><p>ADMIXTURE.jl exports <code>ADMIXTURE_EXE</code>, which points to the installed ADMIXTURE executable. Users can use it to run ADMIXTURE directly. For example,</p><pre><code class="language-julia">run(`$ADMIXTURE_EXE hapmap3/hapmap3.bed 3`)</code></pre><h2 id="References"><a class="docs-heading-anchor" href="#References">References</a><a id="References-1"></a><a class="docs-heading-anchor-permalink" href="#References" title="Permalink"></a></h2><p>The methodology is described in following papers. Please cite if you use ADMIXTURE or ADMIXTURE.jl.</p><ol><li><p>D Alexander, J Novembre, and K Lange. (2009) Fast model-based estimation of ancestry in unrelated individuals, <em>Genome Research</em>, 19(9):1655–1664. <a href="https://dx.doi.org/10.1101/gr.094052.109">https://dx.doi.org/10.1101/gr.094052.109</a>  </p></li><li><p>D Alexander and K Lange. (2011) Enhancements to the ADMIXTURE algorithm for individual ancestry estimation. <em>BMC Bioinformatics</em>, 12:246. <a href="https://doi.org/10.1186/1471-2105-12-246">https://doi.org/10.1186/1471-2105-12-246</a>  </p></li><li><p>S Shringarpure, C Bustamante, K Lange, and D Alexander. (2016) Efficient analysis of large datasets and sex bias with ADMIXTURE. <em>BMC Bioinformatics</em> 17:218. <a href="https://doi.org/10.1186/s12859-016-1082-x">https://doi.org/10.1186/s12859-016-1082-x</a></p></li></ol><pre><code class="language-julia"># clean up the HAPMAP3 files
rm(&quot;hapmap3.3.P&quot;, force = true)
rm(&quot;hapmap3.3.Q&quot;, force = true)
rm(&quot;hapmap3.tar&quot;, force = true)
rm(&quot;hapmap3&quot;, recursive = true)</code></pre></article><nav class="docs-footer"><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> on <span class="colophon-date" title="Wednesday 5 August 2020 15:38">Wednesday 5 August 2020</span>. Using Julia version 1.5.0.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
