var documenterSearchIndex = {"docs":
[{"location":"#ADMIXTURE.jl","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"","category":"section"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"ADMIXTURE.jl is a Julia wrapper of the popular ADMIXTURE program for estimating ancestry in a model-based manner from large autosomal SNP genotype data sets.","category":"page"},{"location":"#Installation","page":"ADMIXTURE.jl","title":"Installation","text":"","category":"section"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"ADMIXTURE.jl requires Julia v1.3 or later. See documentation for usage. It is not yet registered and can be installed, in the Julia Pkg mode, by","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"(@v1.5) Pkg> add https://github.com/OpenMendel/ADMIXTURE.jl","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"The original ADMIXTURE software thus ADMIXTURE.jl only support Linux and MacOS.","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"versioninfo()","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"Julia Version 1.5.0\nCommit 96786e22cc (2020-08-01 23:44 UTC)\nPlatform Info:\n  OS: macOS (x86_64-apple-darwin18.7.0)\n  CPU: Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz\n  WORD_SIZE: 64\n  LIBM: libopenlibm\n  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)\nEnvironment:\n  JULIA_EDITOR = code\n  JULIA_NUM_THREADS = 4","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"# for use in this tutorial\nusing ADMIXTURE, DelimitedFiles, Glob, Tar","category":"page"},{"location":"#Quick-example","page":"ADMIXTURE.jl","title":"Quick example","text":"","category":"section"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"ADMIXTURE.jl only exports one function admixture. The usage resembles the original ADMIXTURE software. ","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"Let's first download a sample data set (in PLINK binary format) to the /hapmap3 folder.","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"isfile(\"hapmap3.tar\") || \ndownload(\"http://dalexander.github.io/admixture/hapmap3-files.tar.gz\", \n         joinpath(pwd(), \"hapmap3.tar\"))\nisdir(\"hapmap3\") ||\nTar.extract(\"hapmap3.tar\", joinpath(pwd(), \"hapmap3\"))\nreaddir(glob\"hapmap3/hapmap3.*\")","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"4-element Array{String,1}:\n \"hapmap3/hapmap3.bed\"\n \"hapmap3/hapmap3.bim\"\n \"hapmap3/hapmap3.fam\"\n \"hapmap3/hapmap3.map\"","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"To estimate ancestry from K=3 populations, simply run","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"P, Q = admixture(\"hapmap3/hapmap3.bed\", 3);","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"****                   ADMIXTURE Version 1.3.0                  ****\n****                    Copyright 2008-2015                     ****\n****           David Alexander, Suyash Shringarpure,            ****\n****                John  Novembre, Ken Lange                   ****\n****                                                            ****\n****                 Please cite our paper!                     ****\n****   Information at www.genetics.ucla.edu/software/admixture  ****\n\nRandom seed: 43\nPoint estimation method: Block relaxation algorithm\nConvergence acceleration algorithm: QuasiNewton, 3 secant conditions\nPoint estimation will terminate when objective function delta < 0.0001\nEstimation of standard errors disabled; will compute point estimates only.\n\n\n┌ Info: ADMIXTURE command:\n│ `/Users/huazhou/.julia/artifacts/7d0732a6f20d0cf8c106525a4602fc824e4cff31/dist/admixture_macosx-1.3.0/admixture /Users/huazhou/.julia/dev/ADMIXTURE/docs/hapmap3/hapmap3.bed 3`\n└ @ ADMIXTURE /Users/huazhou/.julia/dev/ADMIXTURE/src/ADMIXTURE.jl:57\n┌ Info: Output directory: /Users/huazhou/.julia/dev/ADMIXTURE/docs\n└ @ ADMIXTURE /Users/huazhou/.julia/dev/ADMIXTURE/src/ADMIXTURE.jl:58\n\n\nSize of G: 324x13928\nPerforming five EM steps to prime main algorithm\n1 (EM) \tElapsed: 0.239\tLoglikelihood: -4.38757e+06\t(delta): 2.87325e+06\n2 (EM) \tElapsed: 0.239\tLoglikelihood: -4.25681e+06\t(delta): 130762\n3 (EM) \tElapsed: 0.239\tLoglikelihood: -4.21622e+06\t(delta): 40582.9\n4 (EM) \tElapsed: 0.238\tLoglikelihood: -4.19347e+06\t(delta): 22748.2\n5 (EM) \tElapsed: 0.238\tLoglikelihood: -4.17881e+06\t(delta): 14663.1\nInitial loglikelihood: -4.17881e+06\nStarting main algorithm\n1 (QN/Block) \tElapsed: 0.493\tLoglikelihood: -3.94775e+06\t(delta): 231058\n2 (QN/Block) \tElapsed: 0.5\tLoglikelihood: -3.8802e+06\t(delta): 67554.6\n3 (QN/Block) \tElapsed: 0.574\tLoglikelihood: -3.83232e+06\t(delta): 47883.8\n4 (QN/Block) \tElapsed: 0.672\tLoglikelihood: -3.81118e+06\t(delta): 21138.2\n5 (QN/Block) \tElapsed: 0.611\tLoglikelihood: -3.80682e+06\t(delta): 4354.36\n6 (QN/Block) \tElapsed: 0.568\tLoglikelihood: -3.80474e+06\t(delta): 2085.65\n7 (QN/Block) \tElapsed: 0.582\tLoglikelihood: -3.80362e+06\t(delta): 1112.58\n8 (QN/Block) \tElapsed: 0.61\tLoglikelihood: -3.80276e+06\t(delta): 865.01\n9 (QN/Block) \tElapsed: 0.574\tLoglikelihood: -3.80209e+06\t(delta): 666.662\n10 (QN/Block) \tElapsed: 0.679\tLoglikelihood: -3.80151e+06\t(delta): 579.49\n11 (QN/Block) \tElapsed: 0.613\tLoglikelihood: -3.80097e+06\t(delta): 548.156\n12 (QN/Block) \tElapsed: 0.648\tLoglikelihood: -3.80049e+06\t(delta): 473.565\n13 (QN/Block) \tElapsed: 0.571\tLoglikelihood: -3.80023e+06\t(delta): 258.61\n14 (QN/Block) \tElapsed: 0.643\tLoglikelihood: -3.80005e+06\t(delta): 179.949\n15 (QN/Block) \tElapsed: 0.683\tLoglikelihood: -3.79991e+06\t(delta): 146.707\n16 (QN/Block) \tElapsed: 0.625\tLoglikelihood: -3.79989e+06\t(delta): 13.1942\n17 (QN/Block) \tElapsed: 0.681\tLoglikelihood: -3.79989e+06\t(delta): 4.60747\n18 (QN/Block) \tElapsed: 0.585\tLoglikelihood: -3.79989e+06\t(delta): 1.50012\n19 (QN/Block) \tElapsed: 0.583\tLoglikelihood: -3.79989e+06\t(delta): 0.128916\n20 (QN/Block) \tElapsed: 0.578\tLoglikelihood: -3.79989e+06\t(delta): 0.00182983\n21 (QN/Block) \tElapsed: 0.567\tLoglikelihood: -3.79989e+06\t(delta): 4.33787e-05\nSummary: \nConverged in 21 iterations (14.858 sec)\nLoglikelihood: -3799887.171935\nFst divergences between estimated populations: \n\tPop0\tPop1\t\nPop0\t\nPop1\t0.163\t\nPop2\t0.073\t0.156\t\nWriting output files.","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"The P matrix contains estimated allele frequencies of the inferred ancestral populations.","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"P","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"13928×3 Array{Float64,2}:\n 0.99999   0.99999   0.99999\n 0.946581  0.934992  0.901852\n 0.989626  0.382598  0.918612\n 0.973109  0.682057  0.907595\n 0.678695  0.918927  0.129153\n 0.99999   0.99999   0.99999\n 0.99999   0.990119  0.99999\n 0.841989  0.203466  0.851233\n 0.967501  0.86069   0.622157\n 0.870693  0.862778  0.842376\n 0.99999   0.99999   0.99999\n 0.279564  0.830299  0.224164\n 0.777817  0.555361  0.950294\n ⋮                   \n 0.746245  0.769682  0.753071\n 0.7462    0.769957  0.752793\n 0.752064  0.734527  0.753725\n 0.745593  0.769889  0.754917\n 0.749709  0.769826  0.755746\n 0.717428  0.67002   0.702682\n 0.745593  0.769889  0.754917\n 0.745593  0.769889  0.754917\n 0.749937  0.767978  0.755766\n 0.99999   0.996706  0.99999\n 0.978953  0.982035  0.926256\n 0.99999   0.993413  0.992614","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"The Q matrix contains the estimated ancestry fractions.","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"Q","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"324×3 Array{Float64,2}:\n 1.0e-5    0.896321  0.103669\n 0.009659  0.830876  0.159465\n 0.05577   0.725441  0.21879\n 1.0e-5    0.866447  0.133543\n 0.029255  0.88897   0.081775\n 0.009302  0.859576  0.131122\n 1.0e-5    0.715624  0.284366\n 0.013736  0.810352  0.175913\n 1.0e-5    0.727122  0.272868\n 0.03487   0.821125  0.144004\n 1.0e-5    0.886316  0.113674\n 0.011359  0.863532  0.125109\n 1.0e-5    0.856018  0.143972\n ⋮                   \n 1.0e-5    0.99998   1.0e-5\n 1.0e-5    0.99998   1.0e-5\n 1.0e-5    0.99998   1.0e-5\n 1.0e-5    0.99998   1.0e-5\n 1.0e-5    0.99998   1.0e-5\n 1.0e-5    0.99998   1.0e-5\n 1.0e-5    0.99998   1.0e-5\n 1.0e-5    0.99998   1.0e-5\n 1.0e-5    0.99998   1.0e-5\n 1.0e-5    0.99998   1.0e-5\n 1.0e-5    0.99998   1.0e-5\n 0.004387  0.995603  1.0e-5","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"The output files of ADMIXTURE are available in the working directory.","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"readdir(glob\"hapmap3.3.*\")","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"2-element Array{String,1}:\n \"hapmap3.3.P\"\n \"hapmap3.3.Q\"","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"We can read the output files by","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"P = readdlm(\"hapmap3.3.P\", ' ', Float64)\nQ = readdlm(\"hapmap3.3.Q\", ' ', Float64);","category":"page"},{"location":"#Detailed-options","page":"ADMIXTURE.jl","title":"Detailed options","text":"","category":"section"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"The admixture function wraps most options for running ADMIXTURE program. The keywords closely follow those of ADMIXTURE.","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"admixture","category":"page"},{"location":"#ADMIXTURE.admixture","page":"ADMIXTURE.jl","title":"ADMIXTURE.admixture","text":"admixture(inputFile, K; kwargs...)\n\nCalculate the admixture estimates assuming K populations. Input file can be  .bed (together with .bim, binary PLINK format), geno (STRUCTURE format),  or ped (12-coding PLINK format). \n\nKeyword arguments\n\nB: compute standard errors by bootstrap; default is nothing.  \nj: number of threads to be used; default is nothing.\ns: random seed; integer or :time; default is nothing. \nC: major convergence criterion; float or integer; default is nothing. \nc: minor convergence criterion; float or integer; default is nothing.\nqn: secant pairs to be used in quasi-Newton acceleration; default is 3.\nl: tuning parameter in penalized estimation.\ne: tuning parameter in penalized estimation.\n\n\n\n\n\n","category":"function"},{"location":"#Run-ADMIXTURE-directly","page":"ADMIXTURE.jl","title":"Run ADMIXTURE directly","text":"","category":"section"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"ADMIXTURE.jl exports ADMIXTURE_EXE, which points to the installed ADMIXTURE executable. Users can use it to run ADMIXTURE directly. For example,","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"run(`$ADMIXTURE_EXE hapmap3/hapmap3.bed 3`)","category":"page"},{"location":"#References","page":"ADMIXTURE.jl","title":"References","text":"","category":"section"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"The methodology is described in following papers. Please cite if you use ADMIXTURE or ADMIXTURE.jl.","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"D Alexander, J Novembre, and K Lange. (2009) Fast model-based estimation of ancestry in unrelated individuals, Genome Research, 19(9):1655–1664. https://dx.doi.org/10.1101/gr.094052.109  \nD Alexander and K Lange. (2011) Enhancements to the ADMIXTURE algorithm for individual ancestry estimation. BMC Bioinformatics, 12:246. https://doi.org/10.1186/1471-2105-12-246  \nS Shringarpure, C Bustamante, K Lange, and D Alexander. (2016) Efficient analysis of large datasets and sex bias with ADMIXTURE. BMC Bioinformatics 17:218. https://doi.org/10.1186/s12859-016-1082-x","category":"page"},{"location":"","page":"ADMIXTURE.jl","title":"ADMIXTURE.jl","text":"# clean up the HAPMAP3 files\nrm(\"hapmap3.3.P\", force = true)\nrm(\"hapmap3.3.Q\", force = true)\nrm(\"hapmap3.tar\", force = true)\nrm(\"hapmap3\", recursive = true)","category":"page"}]
}
